packages/core/src/UserState.ts

```diff
@@ -211,7 +211,7 @@ export default class UserState {
         const data = Array(this.sync.settings.fieldCount).fill(BigInt(0))
         const orClauses = [] as any[]
         const attesterId = this.sync.attesterId
-        const toEpoch = _toEpoch ?? this.sync.calcCurrentEpoch()
+        const toEpoch = _toEpoch ?? (await this.latestTransitionedEpoch())
```

Comment:
This breaks the functionality of `getData`. [`getData`](https://developer.unirep.io/docs/next/core-api/user-state#getdata) returns all the data a user has received. [`getProvableData`](https://developer.unirep.io/docs/next/core-api/user-state#getprovabledata) returns the provable data.

Reply:
I think it should be `latestTransitionedEpoch`
and it can also be `currentEpoch` if user already transition to the current epoch
so it can make sure the user already transition to this epoch
(if user doesn't transition but receive data, it could be wrong)

e.g. if current epoch is 5 but user's latest transition epoch is 3
and it shouldn't have rep data in epoch 5 if he doesn't transition to 5 
(all data is ~3 epochs and provable data is ~2 epochs)
if he then transitions to 5, it can be valid (all data is ~5 epochs and provable data is ~4 epoch)
if he does not transition to 5, but receive rep data at 5, it should not be valid